javascript/trees/treemax/treemax.js

```diff
@@ -1 +1,36 @@
 'use strict';
+
+class Node {
+  constructor(value, left = null, right = null) {
+    this.value = value;
+    this.left = left;
+    this.right = right;
+  }
+}
+
+class BinaryTree {
+  constructor(root = null) {
+    this.root = new Node(root);
+  }
+
+  findMaximum(node) {
+    if (!this.root) {
+      return null;
+    }
+    let maxValue = null;
+    // could do anything but we're just logging
+    console.log(node.value);
+    if ( node.value > maxValue) {
+      maxValue = node.value;
+    }
+    if (node.left) {
+      this.findMaximum(node.left);
+    }
```

Comment:
You have a scoping problem here. You define `maxValue` then recursively call the function it was defined in. This will create a new variable that is scoped to that function call rather than the original one outside that function. While you could try to pass values around as arguments and returns, it is much easier to use a helper function:

```js
findMaximum() {
  let maxValue
  const traverse = (node) => {
    // comparators, recursive calls, etc.
  }
  traverse(this.root)
  return maxValue
}